pkg/payload/precondition/clusterversion/upgradeable.go

```diff
@@ -111,18 +115,96 @@ func (pf *Upgradeable) Run(ctx context.Context, releaseContext precondition.Rele
 // Name returns Name for the precondition.
 func (pf *Upgradeable) Name() string { return "ClusterVersionUpgradeable" }
 
+//  RecentEtcdBackup checks if a recent etcd backup has been taken.
+type RecentEtcdBackup struct {
+	key      string
+	cvLister configv1listers.ClusterVersionLister
+	coLister configv1listers.ClusterOperatorLister
+}
+
+// NewRecentEtcdBackup returns a new RecentEtcdBackup precondition check.
+func NewRecentEtcdBackup(cvLister configv1listers.ClusterVersionLister, coLister configv1listers.ClusterOperatorLister) *RecentEtcdBackup {
+	return &RecentEtcdBackup{
+		key:      "version",
+		cvLister: cvLister,
+		coLister: coLister,
+	}
+}
+
+func recentEtcdBackupCondition(lister configv1listers.ClusterOperatorLister) (reason string, message string) {
+	var msgDetail string
+	ops, err := lister.Get("etcd")
+	if err == nil {
+		backupCondition := resourcemerge.FindOperatorStatusCondition(ops.Status.Conditions, backupConditionType)
+		if backupCondition == nil {
+			reason = "EtcdRecentBackupNotSet"
+			msgDetail = "etcd backup condition is not set."
+		} else if backupCondition.Status != configv1.ConditionTrue {
+			reason = backupCondition.Reason
+			msgDetail = backupCondition.Message
+		}
+	} else {
+		reason = "UnableToGetEtcdOperator"
+		msgDetail = fmt.Sprintf("Unable to get etcd operator, err=%v.", err)
+	}
+	if len(msgDetail) > 0 {
+		message = fmt.Sprintf("%s: %s", backupConditionType, msgDetail)
+	}
+	return reason, message
+}
+
+// Run runs the RecentEtcdBackup precondition. It returns a PreconditionError until Etcd indicates that a
+// recent etcd backup has been taken.
+func (pf *RecentEtcdBackup) Run(ctx context.Context, releaseContext precondition.ReleaseContext, clusterVersion *configv1.ClusterVersion) error {
+	cv, err := pf.cvLister.Get(pf.key)
+	if apierrors.IsNotFound(err) || meta.IsNoMatchError(err) {
+		return nil
+	}
+	if err != nil {
+		return &precondition.Error{
+			Nested:  err,
+			Reason:  "UnknownError",
+			Message: err.Error(),
+			Name:    pf.Name(),
+		}
+	}
+
+	currentVersion := getCurrentVersion(cv.Status.History)
+	currentMinor := getEffectiveMinor(currentVersion)
+	desiredMinor := getEffectiveMinor(releaseContext.DesiredVersion)
+
+	if minorVersionUpgrade(currentMinor, desiredMinor) {
+		reason, message := recentEtcdBackupCondition(pf.coLister)
+		if len(reason) > 0 {
+			return &precondition.Error{
+				Reason:  reason,
+				Message: message,
+				Name:    pf.Name(),
+			}
+		}
+	}
+	return nil
+}
+
+// Name returns Name for the precondition.
+func (pf *RecentEtcdBackup) Name() string { return "RecentEtcdBackup" }
```

Comment:
If we name this `EtcdRecentBackup`, that name will be added to the `Failing=True` message, and we'll have redundant coverage for the "`message` contains a `RecentBackup` substring" that the etcd operator is looking for.

Reply:
done